### PR TITLE
osd: fix some osd beacon bugs

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by dmick** [comment]
Is there an ||-to-&& change somewhere here I'm not seeing?

**Comment by gregsfortytwo** [comment]
"change this to instead *not* send when monmap epoch==0"
(emphasis mine)

**Comment by liewegas** [comment]
A minimal fix would be to change || to && (that is presumably what was intended), but this is better.

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -5560,15 +5560,17 @@ void OSD::flush_pg_stats()
 
 void OSD::send_beacon(const ceph::coarse_mono_clock::time_point& now)
 {
-  dout(20) << __func__ << dendl;
-  last_sent_beacon = now;
   const auto& monmap = monc->monmap;
   // send beacon to mon even if we are just connected, and the monmap is not
   // initialized yet by then.
-  if (monmap.epoch == 0 &&
+  if (monmap.epoch > 0 &&
```
Comment (tchaikov) [inline src/osd/OSD.cc]: right, `||` was what was intended. maybe we can remove the comment above, which will be confusing then.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-04-02_04:47:42-rados-wip-kefu-testing---basic-mira/
- http://pulpito.ceph.com/kchai-2017-04-02_16:06:02-rados-wip-kefu-testing---basic-mira/

failed test are fixed by #14294 and #14284

**Pull request merged**